crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1141,9 +1151,18 @@ where
                 conflicting_tx_digests: state.conflicting_tx_digests,
             }
         } else {
-            // No conflicting transaction and transaction state is still retryable.
-            AggregatorProcessTransactionError::RetryableTransaction {
-                errors: state.errors,
+            let overloaded_stake = self.get_overloaded_stake(&state);
+            // If at least quorum of the validators are overloaded then we consider the system to be overloaded.
+            if overloaded_stake >= quorum_threshold {
```

Comment:
Is `2f + 1` a reasonable threshold here?

Reply:
Look like this case is similar to object/package not found. Wonder if you should just add the check [here](https://github.com/MystenLabs/sui/blob/204f74ca742b47260dd1caf42a37e5940a512256/crates/sui-core/src/authority_aggregator.rs#L1075-L1082) instead. Exit early and then mark the state as non retryable if you are over quorum threshold?

Reply:
With the current way we deal with object/package not found, we will actually return a `AggregatorProcessTransactionError::FatalTransaction` error. But in the system overload case, the transaction isn't really fatal and we want to return something more meaningful to the user.

Reply:
Up to you, but my suggestion would be to move the following code to [here](https://github.com/MystenLabs/sui/blob/204f74ca742b47260dd1caf42a37e5940a512256/crates/sui-core/src/authority_aggregator.rs#LL1107C9-L1111C10) and then you can mark the tx state as not retryable the moment overload_stake is >= `quorum_threshold`. Best of both worlds 😄 

```
let overloaded_stake = self.get_overloaded_stake(&state);
 // If at least quorum of the validators are overloaded then we consider the system to be overloaded.
 if overloaded_stake >= quorum_threshold {
    AggregatorProcessTransactionError::SystemOverload {
     overloaded_stake,
     errors: state.errors,
}
```

Reply:
I'm not sure we want to mark overloaded transactions as non-retriable, even when a quorum is overloaded.

Reply:
Fair enough, you guys are right and this is not the same case as object/package not found. Not really a nonretryable error

Reply:
I think @arun-koshy what you said makes sense though that we should probably exit the moment overload_stake is >= quorum_threshold, so that we don't end up trying to get responses from everyone when the system is already overloaded. I'll update the PR.